README.md

```diff
@@ -41,45 +41,174 @@ To communicate with CoolWallet device, you need to specify a bluetooth transport
+
+Obtain app key pairs.
+
+```javascript
+const keyPair = crypto.key.generateKeyPair()
+localStorage.setItem('appPublicKey', keyPair.publicKey)
+localStorage.setItem('appPrivateKey', keyPair.privateKey)
+```
+
+- keyPair: The keys use to check your app.
+
+### Register card
+
+Register card and obtain the appId.
+
+```javascript
+const name = 'your app name'
+const SEPublicKey = localStorage.getItem('SEPublicKey')
+const appId = await apdu.pair.register(transport, appPublicKey, password, name, SEPublicKey);
+```
+
+### Create / Recover the wallet
+
+Use function `setSeed` to create or recover your wallet.
+
+```javascript
+const seedHex = bip39.mnemonicToSeedSync(mnemonic).toString('hex');
+await apdu.wallet.setSeed(transport, appId, appPrivateKey, seedHex, SEPublicKey)
+
+```
+
+If you want to create seed by card, you can use function `createSeedByCard`. And also choose the length of seed(12, 18, 24).
+
+```javascript
+await apdu.wallet.createSeedByCard(transport, appId, appPrivateKey, 12);
+```
+
+
+### Integration
+
+```
+npm install @coolwallet/eth
+```
+
+```javascript
+import cwsETH from '@coolwallet/eth'
+
+const ETH = new cwsETH();
+```
+
+### Get Address
+
+```javascript
+const address = await ETH.getAddress(
+  transport,
+  appPrivateKey,
+  appId,
+  addressIdx
+); 
+
+```
+
+The address generated is compatible to BIP44 with **account** and **change** set to 0, which means calling `getAddress(i)` will get the address of folllowing BIP44 path:
+
+```none
+m/44'/60'/0'/0/{i}
+```
+
+In the design of current hardware, we only support path `m/44'/60'/0'/0/{i}` for speed optimization. This might change in the future and we will then open a more general interface to deal with custom path.
+
+If you have `accountPublicKey` and `accountChainCode`, you can use the function `ETH.getAddressByAccountKey()` to get the address. 
+```javascript
+const address = await ETH.getAddressByAccountKey(
+  accountPublicKey,
+  accountChainCode,
+  addressIndex
+);
+```
+
+### Sign Transaction
 
+The signedTx is signed by CoolWallet, which can be sent directly.
 
-## Examples
+```javascript
+const transaction = {
+    nonce: "0x21d",
+    gasPrice: "0x59682f00",
+    gasLimit: "0x5208",
+    to: "0x81bb32e4A7e4d0500d11A52F3a5F60c9A6Ef126C",
+    value: "0x5af3107a4000",
+    data: "0x00",
+    chainId: 1
+};
+const signTxData = {
+  transport,
+  appPrivateKey,
+  appId,
+  transaction,
+  addressIndex,
+};
 
-If you want to build your own app with this sdk, you might find the following repos useful:
+const signedTx = await ETH.signTransaction(signTxData);
 
-- [React Example](https://github.com/antoncoding/cws-web-ble-demo) (with `web-ble`)
+```
 
-- [React-Native Example](https://github.com/kunmingLiu/cws-rn-ble-demo) (with `rn-ble`)
+## Integration
 
-If you build something new, welcome to contact us to put your work in the list.
+If you want to integrate the other coin into CoolWallet, you can refer to [CONTRIBUTING](./CONTRIBUTING.md) or contact us.
```

Comment:
If you're interested to develop new coin for CoolWallet Pro, please see [CONTRIBUTING](./CONTRIBUTING.md) for more information.